store/types/proof.go

```diff
@@ -91,7 +91,7 @@ func (op CommitmentOp) Run(args [][]byte) ([][]byte, error) {
 	// calculate root from proof
 	root, err := op.Proof.Calculate()
 	if err != nil {
-		return nil, sdkerrors.Wrapf(ErrInvalidProof, "could not calculate root for proof: %v", err)
+		return nil, sdkerrors.Wrapf(ErrInvalidProof, "could not calculate root for proof: %w", err)
```

Comment:
I think this is technically what you want:

```suggestion
		return nil, sdkerrors.Wrapf(err, "could not calculate root for proof)
```

If you want to preserve `err` AND `ErrInvalidProof`, then you'll have to double-wrap.